alexandria/data_structures/src/stack.cairo

```diff
@@ -14,98 +14,150 @@
 //! ```
 
 // Core lib imports
-use array::ArrayTrait;
+use dict::Felt252DictTrait;
 use option::OptionTrait;
+use traits::Into;
+use nullable::NullableTrait;
 
-use alexandria_data_structures::array_slice;
-
-const ZERO_USIZE: usize = 0;
-
-#[derive(Drop)]
-struct Stack {
-    elements: Array<u256>, 
-}
-
-trait StackTrait {
+trait StackTrait<S, T> {
     /// Creates a new Stack instance.
-    fn new() -> Stack;
+    fn new() -> S;
     /// Pushes a new value onto the stack.
-    fn push(ref self: Stack, value: u256);
+    fn push(ref self: S, value: T);
     /// Removes the last item from the stack and returns it, or None if the stack is empty.
-    fn pop(ref self: Stack) -> Option<u256>;
+    fn pop(ref self: S) -> Option<T>;
     /// Returns the last item from the stack without removing it, or None if the stack is empty.
-    fn peek(self: @Stack) -> Option<u256>;
+    fn peek(ref self: S) -> Option<T>;
     /// Returns the number of items in the stack.
-    fn len(self: @Stack) -> usize;
+    fn len(self: @S) -> usize;
     /// Returns true if the stack is empty.
-    fn is_empty(self: @Stack) -> bool;
+    fn is_empty(self: @S) -> bool;
+}
+
+struct Felt252Stack<T> {
```

Comment:
Is there a reason for name this `Felt252Stack` as it can take any generic T?
Maybe it makes sense to rename it to something more generic?

Reply:
Name reflects the fact that it accepts types of felt size. Any suggestions for a better name?

Reply:
Thanks for the clarification, then it is fine as it matches with other standards (HashMap, ...)